From 87ac2495700918483306e8e61175ec784e3af8aa Mon Sep 17 00:00:00 2001
From: No767 <73260931+No767@users.noreply.github.com>
Date: Fri, 4 Feb 2022 23:55:12 -0800
Subject: [PATCH] Migrate from Pycord v1.7.3 to v2.0.0b3

Had some issues since v2 was almost a full rewrite of Pycord
---
 Bot/Cogs/advice.py                |  2 +-
 Bot/Cogs/bot-info.py              |  6 ++---
 Bot/Cogs/global-error-handling.py |  8 -------
 Bot/Cogs/global.py                |  2 +-
 Bot/Cogs/rinhelp.py               | 38 +++++++++++++++----------------
 Bot/Cogs/rininfo.py               |  8 +++----
 Bot/Cogs/rininvite.py             |  2 +-
 Bot/Cogs/translate.py             |  6 ++---
 Bot/Cogs/waifu-pics.py            |  4 ++--
 Bot/rinbot.py                     |  2 +-
 10 files changed, 35 insertions(+), 43 deletions(-)

diff --git a/Bot/Cogs/advice.py b/Bot/Cogs/advice.py
index 8d1fdf4..07a8227 100644
--- a/Bot/Cogs/advice.py
+++ b/Bot/Cogs/advice.py
@@ -24,7 +24,7 @@ class advice_slip(commands.Cog):
                     embedVar.description = f"{advice_slip_formatted['slip']['advice']}"
                     embedVar.set_footer(
                         text=f"Requested by {ctx.message.author.name}",
-                        icon_url=ctx.message.author.avatar_url,
+                        icon_url=ctx.message.author.avatar,
                     )
                     await ctx.send(embed=embedVar)
                 except Exception as e:
diff --git a/Bot/Cogs/bot-info.py b/Bot/Cogs/bot-info.py
index df65e67..7976ab3 100644
--- a/Bot/Cogs/bot-info.py
+++ b/Bot/Cogs/bot-info.py
@@ -15,7 +15,7 @@ def fast_embed(content):
     return discord.Embed(description=content, color=discord_colors())
 
 
-class Utility(commands.Cog):
+class botInfo(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
 
@@ -38,11 +38,11 @@ class Utility(commands.Cog):
             value=average_members_per_guild,
             inline=False,
         )
-        embed.set_thumbnail(url=bot.user.avatar_url)
+        embed.set_thumbnail(url=bot.user.display_avatar)
         await ctx.send(embed=embed)
 
     asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
 
 
 def setup(bot):
-    bot.add_cog(Utility(bot))
+    bot.add_cog(botInfo(bot))
diff --git a/Bot/Cogs/global-error-handling.py b/Bot/Cogs/global-error-handling.py
index 29e34c1..8de6223 100644
--- a/Bot/Cogs/global-error-handling.py
+++ b/Bot/Cogs/global-error-handling.py
@@ -61,14 +61,6 @@ class everyonePingChecker(commands.Cog):
 
     asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
 
-
-# Remove this again due to discord.bots.gg testing
-# if isinstance(error, commands.CommandNotFound):
-#     embedVar = discord.Embed(
-#         color=discord.Color.from_rgb(226, 199, 255))
-#     embedVar.description = f"{error}. Please try again, or refer to either `rinhelp` or the [docs](https://rin-docs.readthedocs.io/en/latest)"
-#     msg = await ctx.send(embed=embedVar, delete_after=10)
-#     await msg.delete(delay=10)
 def setup(bot):
     bot.add_cog(ErrorHandler(bot))
     bot.add_cog(everyonePingChecker(bot))
diff --git a/Bot/Cogs/global.py b/Bot/Cogs/global.py
index 26a1e25..cb51f46 100644
--- a/Bot/Cogs/global.py
+++ b/Bot/Cogs/global.py
@@ -20,7 +20,7 @@ class Global(commands.Cog):
             message_embed = discord.Embed()
             message_embed.set_footer(
                 text=f"Sent by {message.author.name} at {message.author.guild.name}",
-                icon_url=message.author.avatar_url,
+                icon_url=message.author.display_avatar,
             )
 
             # if image as link
diff --git a/Bot/Cogs/rinhelp.py b/Bot/Cogs/rinhelp.py
index 8e18fe8..a61f048 100644
--- a/Bot/Cogs/rinhelp.py
+++ b/Bot/Cogs/rinhelp.py
@@ -61,7 +61,7 @@ class rinhelp(commands.Cog):
                     name="OpenAI", value="`.rinhelp openai`", inline=True
                 )
                 embedVar.set_author(
-                    name="Rin Help", icon_url=bot.user.avatar_url)
+                    name="Rin Help", url=discord.Embed.Empty, icon_url=bot.user.display_avatar)
                 embedVar.set_footer(
                     text='Remember, the command prefix for this bot is "."'
                 )
@@ -90,7 +90,7 @@ class rinhelp(commands.Cog):
                     name="`ping`", value="Checks the ping for the bot", inline=True
                 )
                 embedVar.set_author(
-                    name="Rin Help - Admin", icon_url=bot.user.avatar_url
+                    name="Rin Help - Admin", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -108,7 +108,7 @@ class rinhelp(commands.Cog):
                     inline=True,
                 )
                 embedVar.set_author(
-                    name="Rin Help - Twitter", icon_url=bot.user.avatar_url
+                    name="Rin Help - Twitter", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -129,7 +129,7 @@ class rinhelp(commands.Cog):
                     inline=True,
                 )
                 embedVar.set_author(
-                    name="Rin Help - Reddit", icon_url=bot.user.avatar_url
+                    name="Rin Help - Reddit", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -187,7 +187,7 @@ class rinhelp(commands.Cog):
                     inline=True,
                 )
                 embedVar.set_author(
-                    name="Rin Help - Minecraft", icon_url=bot.user.avatar_url
+                    name="Rin Help - Minecraft", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -221,7 +221,7 @@ class rinhelp(commands.Cog):
                     inline=True,
                 )
                 embedVar.set_author(name="Rin Help - Fun",
-                                    icon_url=bot.user.avatar_url)
+                                    icon_url=bot.user.display_avatar)
                 await ctx.send(embed=embedVar)
 
             if str(search) in ("instagram", "ig"):
@@ -249,7 +249,7 @@ class rinhelp(commands.Cog):
                     text="Note: There is a 20 sec cooldown for each command. This is to prevent rate limiting."
                 )
                 embedVar.set_author(
-                    name="Rin Help - Instagram", icon_url=bot.user.avatar_url
+                    name="Rin Help - Instagram", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -266,7 +266,7 @@ class rinhelp(commands.Cog):
                     value="Clears number of messages specified from the channel in which in the command was called",
                 )
                 embedVar.set_author(
-                    name="Rin Help - Chat", icon_url=bot.user.avatar_url
+                    name="Rin Help - Chat", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -305,7 +305,7 @@ class rinhelp(commands.Cog):
                     name="`uptime`", value="Checks for Rin's Uptime", inline=True
                 )
                 embedVar.set_author(
-                    name="Rin Help - Misc", icon_url=bot.user.avatar_url
+                    name="Rin Help - Misc", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -341,7 +341,7 @@ class rinhelp(commands.Cog):
                     text='These cmds have the prefix of "deviantart". This prefix can be swapped out for "da" instead. For example, the "deviantart-item" cmd can be shorten to "da-item"'
                 )
                 embedVar.set_author(
-                    name="Rin Help - Deviantart", icon_url=bot.user.avatar_url
+                    name="Rin Help - Deviantart", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -359,7 +359,7 @@ class rinhelp(commands.Cog):
                     inline=True,
                 )
                 embedVar.set_author(
-                    name="Rin Help - Anime", icon_url=bot.user.avatar_url
+                    name="Rin Help - Anime", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -395,7 +395,7 @@ class rinhelp(commands.Cog):
                     text='Also note that the prefix can be shorten down to "jk"'
                 )
                 embedVar.set_author(
-                    name="Rin Help - MyAnimeList/Jikan", icon_url=bot.user.avatar_url
+                    name="Rin Help - MyAnimeList/Jikan", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -413,7 +413,7 @@ class rinhelp(commands.Cog):
                     inline=True,
                 )
                 embedVar.set_author(
-                    name="Rin Help - Topgg", icon_url=bot.user.avatar_url
+                    name="Rin Help - Topgg", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -437,11 +437,11 @@ class rinhelp(commands.Cog):
                     text='The Pinterest cmds also have aliases just like Jikan and DeviantArt. The alias prefix is "pt". Note that the Pinterest API only supports getting info about the user that it is logged into, which means it is grabbing info from my own account. It is not recommended to use this service.'
                 )
                 embedVar.set_author(
-                    name="Rin Help - Pinterest", icon_url=bot.user.avatar_url
+                    name="Rin Help - Pinterest", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
-            if str(search) == "youtube":
+            if str(search) in ["youtube", "yt"]:
                 bot = self.bot
                 embedVar = discord.Embed(color=14414079)
                 embedVar.add_field(
@@ -473,7 +473,7 @@ class rinhelp(commands.Cog):
                     text='Note that the alias prefix is "yt". This means that for example, the cmd "youtube-search" can be shorten down to "yt-search"'
                 )
                 embedVar.set_author(
-                    name="Rin Help - YouTube", icon_url=bot.user.avatar_url
+                    name="Rin Help - YouTube", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -514,7 +514,7 @@ class rinhelp(commands.Cog):
                     inline=True,
                 )
                 embedVar.set_author(
-                    name="Rin Help - Tenor", icon_url=bot.user.avatar_url
+                    name="Rin Help - Tenor", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -537,7 +537,7 @@ class rinhelp(commands.Cog):
                     inline=True,
                 )
                 embedVar.set_author(
-                    name="Rin Help - OpenAI", icon_url=bot.user.avatar_url
+                    name="Rin Help - OpenAI", icon_url=bot.user.display_avatar
                 )
                 await ctx.send(embed=embedVar)
 
@@ -546,7 +546,7 @@ class rinhelp(commands.Cog):
             embedVar = discord.Embed(title="Rin Help", color=14414079)
             embedVar.description = "The query failed."
             embedVar.add_field(name="Error", value=e, inline=True)
-            embedVar.set_thumbnail(url=bot.user.avatar_url)
+            embedVar.set_thumbnail(url=bot.user.display_avatar)
             await ctx.send(embed=embedVar)
 
     asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
diff --git a/Bot/Cogs/rininfo.py b/Bot/Cogs/rininfo.py
index dcc5420..371457d 100644
--- a/Bot/Cogs/rininfo.py
+++ b/Bot/Cogs/rininfo.py
@@ -9,14 +9,14 @@ class info(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
 
-    @commands.command(name="rininfo", help="Server Info")
-    async def on_message(self, ctx):
+    @commands.command(name="rininfo", aliases=["info"])
+    async def rinInfo(self, ctx):
         bot = self.bot
         embedVar = discord.Embed(color=14414079)
-        embedVar.set_author(name="Rin Info", icon_url=bot.user.avatar_url)
+        embedVar.set_author(name="Rin Info", icon_url=bot.user.display_avatar)
         embedVar.add_field(
             name="About",
-            value="Rin is a discord bot which supports obtaining data from third party services such as DeviantArt, Hypixel, Reddit, MyAnimeList/Jikan, and many others. And you can request such data just from Discord. More services are planned to be supported, such as Twitch, and many others. (The MangaDex service is planned for you to be able to read manga from Discord). If you are interested in a general-purpose version of Rin, check out Kumiko",
+            value="Rin is a Discord bot focused on providing data from third party services such as DeviantArt, Hypixel, Reddit, MyAnimeList/Jikan, and many others. Rin allows you to find memes on Reddit with the Reddit service, or get info about your favorite anime with the MAL service. If you want a multipurpose version of Rin, check out Kumiko.",
             inline=False,
         )
         embedVar.add_field(
diff --git a/Bot/Cogs/rininvite.py b/Bot/Cogs/rininvite.py
index 7fec105..3ecb861 100644
--- a/Bot/Cogs/rininvite.py
+++ b/Bot/Cogs/rininvite.py
@@ -14,7 +14,7 @@ class InviteV1(commands.Cog):
         bot = self.bot
         embedVar = discord.Embed()
         embedVar.description = "[Top.gg](https://top.gg/bot/865883525932253184/invite)\n[Fallback URL](https://discord.com/api/oauth2/authorize?client_id=865883525932253184&permissions=150055930992&scope=bot)"
-        embedVar.set_author(name="Invite", icon_url=bot.user.avatar_url)
+        embedVar.set_author(name="Invite", icon_url=bot.user.display_avatar)
         await ctx.send(embed=embedVar)
 
     asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
diff --git a/Bot/Cogs/translate.py b/Bot/Cogs/translate.py
index 0050f25..fbba835 100644
--- a/Bot/Cogs/translate.py
+++ b/Bot/Cogs/translate.py
@@ -6,7 +6,7 @@ from deep_translator import GoogleTranslator
 from discord.ext import commands
 
 
-class Utility(commands.Cog):
+class utils(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
 
@@ -32,7 +32,7 @@ class Utility(commands.Cog):
             translate_embed = discord.Embed(description=translatev2)
             translate_embed.set_footer(
                 text=f"Requested by {ctx.message.author.name}",
-                icon_url=ctx.message.author.avatar_url,
+                icon_url=ctx.message.author.display_avatar,
             )
             await ctx.send(embed=translate_embed)
         except Exception as e:
@@ -42,4 +42,4 @@ class Utility(commands.Cog):
 
 
 def setup(bot):
-    bot.add_cog(Utility(bot))
+    bot.add_cog(utils(bot))
diff --git a/Bot/Cogs/waifu-pics.py b/Bot/Cogs/waifu-pics.py
index e322bb6..9363c5e 100644
--- a/Bot/Cogs/waifu-pics.py
+++ b/Bot/Cogs/waifu-pics.py
@@ -8,7 +8,7 @@ import uvloop
 from discord.ext import commands
 
 
-class waifu(commands.Cog):
+class waifuPics(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
 
@@ -63,4 +63,4 @@ class waifu(commands.Cog):
 
 
 def setup(bot):
-    bot.add_cog(waifu(bot))
+    bot.add_cog(waifuPics(bot))
diff --git a/Bot/rinbot.py b/Bot/rinbot.py
index 7105eea..f2aeb04 100644
--- a/Bot/rinbot.py
+++ b/Bot/rinbot.py
@@ -38,7 +38,7 @@ initial_extensions = [
     "Cogs.uptime",
     "Cogs.mangadex",
     "Cogs.bot-info",
-    "Cogs.openai-gpt3"
+    "Cogs.openai-gpt3",
 ]
 for extension in initial_extensions:
     bot.load_extension(extension)
-- 
2.34.1

